src/lib.rs

```diff
@@ -45,15 +45,57 @@ pub struct Header {
     pub generator: u32,
 }
 
+/// Forces early fragment tests. See https://www.khronos.org/opengl/wiki/Early_Fragment_Test.
+/// In GLSL: layout(early_fragment_tests) in;
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct EarlyFragmentTest {}
+///
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum FragmentDepthWrite {
+    ///
+    Greater,
+
+    ///
+    Less,
+
+    ///
+    Unchanged,
```

Comment:
nit: `Any` would be more appropriate, I think.
It's not like the `Greater` or `Less`  are supposed to *change* the write. They are supposed to indicate what the write already does.

Reply:
Any == None
Unchanged == Some(Unchaged)

Reply:
what's the difference between `None` and `Unchanged`?

Reply:
Here is a GLSL with None(Any): http://shader-playground.timjones.io/b33d1f8e9461265f2ed6c6f8a9151710. You can comment the ```layout``` and It will produce the same thing: with only ExecutionMode DepthReplacing.
Here is a GLSL with Unchanged: http://shader-playground.timjones.io/b0acc9ce04760875c6242fc798aed3d1. You get no ExecutionMode DepthReplacing and this one should be evaluated (at runtime) as incorrect.

The ARB extension of conservative depth states:
```
// assume it may be modified in any way
 layout (depth_any) out float gl_FragDepth;

// assume it will not be modified
layout (depth_unchanged) out float gl_FragDepth;
```

With None/Any you are making no guarantees about gl_FragDepth modification whereas with Unchanged you are explicitly stating you won't modify gl_FragDepth.